src/controllers/balancesController.ts

```diff
@@ -0,0 +1,119 @@
+import { JsonController, Get, Param, QueryParam, BadRequestError, Post, Delete, HttpCode, OnNull, OnUndefined, Res, Ctx } from "routing-controllers";
+import { BalanceRepository, BalanceEntity } from "../domain/balances";
+import { AssetRepository } from "../domain/assets";
+import { EosService } from "../services/eosService";
+import { ConflictError } from "../errors/conflictError";
+import { BlockchainError } from "../errors/blockchainError";
+
+@JsonController("/balances")
+export class BalancesController {
+
+    constructor(
+        private assetRepository: AssetRepository,
+        private balanceRepository: BalanceRepository,
+        private eosService: EosService) {
+    }
+
+    @Get()
+    async balances(
+        @QueryParam("take", { required: true }) take: number,
+        @QueryParam("continuation") continuation?: string) {
+
+        if (Number.isNaN(take) || !Number.isInteger(take) || take <= 0) {
```

Comment:
Just for future updates: Move the take param validation to separate function and use it in all endpoints